lib/akamai_rspec/matchers/caching.rb

```diff
@@ -6,14 +6,20 @@ module AkamaiRSpec
   module Matchers
     define :be_cacheable do |request_count: 4, headers: {}, allow_refresh: false|
       match do |url|
+        @responses = []
+        fail("URL must be a string") unless url.is_a? String
+
         @responses = (1..request_count).map {
           AkamaiRSpec::Request.get url, headers
         }
 
         @responses.any? do |response|
           fail("Error fetching #{url}: #{response}") if response.code != 200
-          return allow_refresh if refresh_hit? response
```

Comment:
This also fixes an issue where this `return` was leaving the entire `match` block!